e2etests/web/regular_integration_tests/lib/screenshot_support.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' as io;
+
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:integration_test/integration_test_driver_extended.dart' as test;
+
+import 'package:web_test_utils/goldens.dart';
+import 'package:web_test_utils/image_compare.dart';
+
+import 'package:image/image.dart';
+
+/// Tolerable pixel difference ratio between the goldens and the screenshots.
+///
+/// We are allowing a higher difference rate compared to the unit tests (where
+/// this rate is set to 0.28), since during the end to end tests there are
+/// more components on the screen which are not related to the functionality
+/// under test ex: a blinking cursor.
+const double kMaxDiffRateFailure = 0.5 / 100; // 0.5%
+
+/// Used for calling `integration_test` package.
+///
+/// Compared to other similar classes which only included the following call:
+/// ```
+/// Future<void> main() async => test.integrationDriver();
+/// ```
+///
+/// this method is able to take screenshot.
+///
+/// It provides an `onScreenshot` callback to the `integrationDriver` method.
+/// It also includes options for updating the golden files.
+Future<void> runTestWithScreenshots(
+    {double diffRateFailure = kMaxDiffRateFailure}) async {
+  final WebFlutterDriver driver =
+      await FlutterDriver.connect() as WebFlutterDriver;
+
+  // Learn the browser in use from the webDriver.
+  final String browser = driver.webDriver.capabilities['browserName'] as String;
+
+  bool updateGoldens = false;
+  final String updateGoldensFlag = io.Platform.environment['UPDATE_GOLDENS'];
```

Comment:
We should validate that this is either `null`, `true`, or `false`. Any other value should raise an exception. Since this can be manually supplied it's not hard to type `ture` or `fasle` by accident.

Reply:
done.